src/features/CheckoutPage/components/ShippingInformation.jsx

```diff
@@ -0,0 +1,97 @@
+
+import React, {useState} from 'react';
+import {fetchInfoFromPincode} from '../../../common/utils/api';
+import PropTypes from 'prop-types';
+
+const ShippingInformation = ({submitHandler}) => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [address, setAddress] = useState('');
+  const [apartment, setApartment] = useState('');
+  const [state, setState] = useState('');
+  const [city, setCity] = useState('');
+  const [pincode, setPincode] = useState('');
+  const formActionMappings = {
+    'firstName': setFirstName,
+    'lastName': setLastName,
+    'address': setAddress,
+    'apartment': setApartment,
+    'state': setState,
+    'city': setCity,
+    'pincode': setPincode,
+  };
+
+  const onFormInputChange = async (e) => {
+    formActionMappings[e.target.name](e.target.value);
+    if (e.target.name == 'pincode') {
+      const response = await fetchInfoFromPincode(e.target.value);
+      if (response[0].Status == 'Success') {
+        setState(response[0]['PostOffice'][0]['State']);
```

Comment:
you can never be sure what's coming in the response. add optinal chaining

Reply:
done